powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/service/v2/SecureVaultService.java

```diff
@@ -89,8 +89,9 @@ public VaultUnlockResponse vaultUnlock(String signatureHeader,
             // Validate the header
             try {
                 PowerAuthSignatureHttpHeaderValidator.validate(header);
-            } catch (InvalidPowerAuthHttpHeaderException e) {
-                throw new PowerAuthAuthenticationException(e.getMessage());
+            } catch (InvalidPowerAuthHttpHeaderException ex) {
+                logger.warn("Signature validation failed, error: {}", ex.getMessage());
+                throw new PowerAuthAuthenticationException("POWER_AUTH_SIGNATURE_INVALID");
```

Comment:
Let's pass `ex` as a parameter to the new exception to keep the cause somewhere.